torchtune/modules/transformer.py

```diff
@@ -500,26 +495,23 @@ def _validate_inputs(
                 f"than max_seq_len ({self.max_seq_len})"
             )
 
-        if self.decoder_caches_are_enabled():
+        if self.caches_are_enabled():
             if mask is None:
                 raise ValueError(
                     "KV-caches for self-attention layers are setup for inference mode, causal masks must be provided!"
                     " Use the `mask` arg to provide a causal mask."
                 )
 
-        if self.encoder_caches_are_enabled():
-            if encoder_mask is None:
+            if encoder_input is None and encoder_mask:
```

Comment:
in there some good default for encoder_mask, like causal mask? If not, just mark as resolved

Reply:
I plan on removing / re-working most of this validation logic to have better defaults, but that will be a follow-up PR.